src/main/fc/runtime_config.h

```diff
@@ -27,9 +27,9 @@ typedef enum {
 
 extern uint8_t armingFlags;
 
-#define DISABLE_ARMING_FLAG(mask) (armingFlags &= ~(mask))
-#define ENABLE_ARMING_FLAG(mask) (armingFlags |= (mask))
-#define ARMING_FLAG(mask) (armingFlags & (mask))
+#define DISABLE_ARMING_FLAG(mask) (armingFlags &= ~((uint8_t)mask))
+#define ENABLE_ARMING_FLAG(mask) (armingFlags |= ((uint8_t)mask))
+#define ARMING_FLAG(mask) (armingFlags & ((uint8_t)mask))
```

Comment:
With these casts we should have some build time checking in `runtime_config.c`, eg
```
BUILD_BUG_ON(sizeof(armingFlags) != sizeof(uint8_t));
BUILD_BUG_ON(sizeof(flightModeFlags) != sizeof(uint16_t));
```
With a comment saying what to do if the build bug occurs.